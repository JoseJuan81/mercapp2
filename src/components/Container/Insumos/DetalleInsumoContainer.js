import React from 'react';
import { useSelector } from 'react-redux';

import { PaginaDetalleInsumo } from '../../../Pages/Insumos/PaginaDetalleInsumo';
import { PaginaLoadingDetalleInsumo } from '../../../Pages/loading/PaginaLoadingDetalleInsumo';

import { DetalleInsumoMenuMobile } from '../../Menus/DetalleInsumoMenuMobile';

export const DetalleInsumoContainer = () => {

    // ===== STORE =====
    const { loading } = useSelector( store => store.loading );
    
    return (
        <div>
            {loading
            ? <PaginaLoadingDetalleInsumo />
            : <PaginaDetalleInsumo />}

            <DetalleInsumoMenuMobile />
        </div>
    )
}
