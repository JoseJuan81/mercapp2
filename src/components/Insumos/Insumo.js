import React from 'react';

const InsumoEtiquetas = ({ labels, checked }) => {
    return (
        <ul className="flex flex-wrap mt-2">

            {
                labels.map(la => (
                    <li
                        className={`
                            bg-white
                            text-xs ${ checked ? 'text-lime-700' : 'text-warmGray-800' }
                            mr-1 mb-1 p-1
                        `}
                    >{ la }</li>
                ))
            }

        </ul>
    )
}

export const Insumo = ({ title, price, currency, labels, checked }) => {

    return (
        <div
            className={`
                duration-200
                rounded-lg ${checked && 'shadow-xl'}
                border border-solid ${checked ? 'border-lime-400' : 'border-warmGray-300'}
            `}
        >

            <dl className="flex flex-auto p-2 mb-2">

                <div className="flex flex-auto">
                    <dt
                        className={`
                            duration-200
                            flex-auto
                            text-2xl font-bold ${checked ? 'text-lime-600' : 'text-warmGray-400'}
                        `}
                    >
                        { title }
                    </dt>

                    <dt
                        className={`
                            text-2xl ${checked ? 'text-lime-400' : 'text-warmGray-400'}
                            ${checked && 'font-bold'}
                        `}
                    >
                        <small
                            className="text-xs font-light"
                        >
                            { currency }
                        </small>
                        <span>{ price }</span>
                    </dt>
                </div>

                <div className="ml-4">...</div>

            </dl>

            <div
                className={`
                    rounded-br-lg rounded-bl-lg
                    ${ checked ? 'bg-lime-100' : 'bg-warmGray-100' }
                    px-2 py-2
                `}
            >
                <InsumoEtiquetas
                    labels={ labels }
                    checked={ checked }
                />
            </div>
            
        </div>
    )
}
