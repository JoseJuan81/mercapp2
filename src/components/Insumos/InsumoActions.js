import React, { useCallback, useState } from 'react'
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import { startDeletingInsumos } from '../../actions/insumosAction';
import { setInsumoToUpdate } from '../../actions/newInsumoAction';
import { removeInsumoFromPurchase } from '../../actions/newPurchaseAction';

import { editarInsumoPath } from '../../constant/routes';
import { EditButton, LeftCircleButton, SeeDetailsButton, TrashButton } from '../Buttons/AppButtons';

export const InsumoBaseActions = React.memo( ({ id }) => {

    // ===== NAVIGATION =====
    const history = useHistory();

    // ===== STORE =====
    const dispatch = useDispatch();

    // ===== STATE =====
    const [toogle, setToogle] = useState(false);

    // ===== FUNCIONES PROPIEAS =====
    const handleClick = useCallback(( ev ) => {
        ev.stopPropagation();

        setToogle( s => !s );
    },[])

    const handleDeleteInsumo = useCallback(( ev ) => {
        ev.stopPropagation();

        dispatch( startDeletingInsumos( id ) );
        setToogle( false );
    },[])
    
    const handleUpdateInsumo = ( ev ) => {
        ev.stopPropagation();

        dispatch( setInsumoToUpdate( id ) );

        history.push( `${ editarInsumoPath }/${ id }` );
    }

    return (
        <div
            className={`
                absolute right-1
                h-10
                bg-warmGray-100
                flex
                transform ${ toogle ? 'translate-x-1' : 'translate-x-40' }
                border border-solid ${ toogle ? 'border-warmGray-200' : 'border-white' }
                rounded-l-full
                duration-300
            `}
        >
            <LeftCircleButton
                isButton
                className={`
                    bg-white
                    flex items-center justify-center
                    text-warmGray-600 text-xl
                    transform ${toogle ? 'rotate-180' : 'rotate-0'}
                    w-10
                    ${ toogle ? 'rounded-full' : 'rounded-l-full' }
                    border border-solid border-warmGray-200
                    duration-300
                `}
                onClick= { handleClick }
            />

            <div
                className="
                    grid grid-cols-3 gap-2
                    px-2
                "
            >
                <SeeDetailsButton
                    className="
                        px-2
                        text-xl text-warmGray-800
                    "
                    to=""
                />

                <EditButton
                    isButton
                    className="
                        px-2
                        text-xl text-warmGray-800
                    "
                    onClick={ handleUpdateInsumo }
                />

                <TrashButton
                    isButton
                    type="button"
                    className="
                        px-2
                        text-xl text-warmGray-800
                    "
                    onClick={ handleDeleteInsumo }
                />
            </div>

        </div>
    )
})

export const InsumoActions = React.memo( ({ id }) => {
    
    // ===== NAVIGATION =====
    const history = useHistory();

    // ===== STORE =====
    const dispatch = useDispatch();

    // ===== STATE =====
    const [toogle, setToogle] = useState(false);

    // ===== FUNCIONES PROPIEAS =====
    const handleClick = useCallback(( ev ) => {
        ev.stopPropagation();

        setToogle( s => !s );
    },[])

    const handleDeleteInsumo = useCallback(( ev ) => {
        ev.stopPropagation();

        dispatch( removeInsumoFromPurchase( id ) );
    },[])
    
    // const handleUpdateInsumo = useCallback(( ev ) => {
    //     ev.stopPropagation();

    //     dispatch( setInsumoToUpdate( id ) );

    //     history.push( `${ editarInsumoPath }/${ id }` );
    // },[])

    return (
        <div
            className={`
                absolute right-1
                h-10
                bg-warmGray-100
                flex
                transform ${ toogle ? 'translate-x-10' : 'translate-x-24' }
                border border-solid ${ toogle ? 'border-warmGray-200' : 'border-white' }
                rounded-l-full
                duration-300
            `}
        >
            <LeftCircleButton
                isButton
                className={`
                    bg-white
                    flex items-center justify-center
                    text-warmGray-600 text-xl
                    transform ${toogle ? 'rotate-180' : 'rotate-0'}
                    w-10
                    ${ toogle ? 'rounded-full' : 'rounded-l-full' }
                    border border-solid border-warmGray-200
                    duration-300
                `}
                onClick= { handleClick }
            />

            <div
                className="
                    grid grid-cols-2 gap-2
                    px-2
                "
            >
                {/* <EditButton
                    isButton
                    className="
                        px-2
                        text-xl text-warmGray-800
                    "
                    onClick={ handleUpdateInsumo }
                /> */}
                <TrashButton
                    isButton
                    type="button"
                    className="
                        px-2
                        text-xl text-warmGray-800
                    "
                    onClick={ handleDeleteInsumo }
                />
            </div>

        </div>
    )
})
