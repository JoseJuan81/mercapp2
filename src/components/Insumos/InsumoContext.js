import { createContext } from 'react';

export const InsumoContext = createContext(null);