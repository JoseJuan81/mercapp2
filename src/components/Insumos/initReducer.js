import { decide, equality, find, reduce } from 'functionallibrary';

import { data as dataInsumos } from '../../fakeData/insumos.json';

import { getFromLocalStorage } from '../../helper/localStorage';
import { alphabeticSorting } from '../../helper/alphabeticSorting';

export const initReducer = () => {
    const dataSelected = getFromLocalStorage('selected-insumos');

    if (!dataSelected || dataSelected.length === 0) {
        return alphabeticSorting(dataInsumos);
    }
    
    const reduceManage = (list, item) => {

        const isChecked = find( equality('title', item.title), dataSelected );

        const addChecked = () => list.concat(isChecked);
        const addNotChecked = (el) => list.concat(el);

        return decide(!!isChecked, addChecked, addNotChecked, item);

    }

    const updatedInsumos = reduce(reduceManage, [], dataInsumos);
    return alphabeticSorting(updatedInsumos);
}
