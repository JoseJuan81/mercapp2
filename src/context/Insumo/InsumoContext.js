import { createContext } from 'react';

/**
 * Crear contexto de Insumo
 */
export const InsumoContext = createContext(null);
