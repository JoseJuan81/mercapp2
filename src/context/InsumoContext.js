import { createContext, useEffect, useMemo, useState } from 'react';
import { removeItemFromArrayByProp } from 'functionallibrary';

import { useIndexDB } from '../hooks/useIndexDB';

import { getFromLocalStorage, setInLocalStorage } from '../helper/localStorage';
import {
    initialInsumos,
    onFilter,
    onSearch,
    updateItemInArrayById,
    selectAllInsumos,
    unSelectAllInsumos,
    updateTotal,
    updatingQuantity,
} from '../helper/helperInsumoContext';

/**
 * Crear contexto de Insumo
 */
export const InsumoContext = createContext(null);

const dataSelected = getFromLocalStorage('selected-insumos') || [];

/**
 * @description funcion wrapper del contexto
 * @param {ReactComponent} children
 */
export const InsumoStore = ({ children }) => {

    // obtener insumos de indexDB
    const {
        deleteInsumo,
        insumos: insumosInLocalBD,
        setNewInsumo
    } = useIndexDB();


    // Ordenar insumos provenientes de indexDB y actualizarlos con los seleccionados
    const initInsumos = useMemo( () => initialInsumos( insumosInLocalBD ), [insumosInLocalBD] );

    const [insumos, setInsumos] = useState(initInsumos);
    const [selectedInsumos, setSelectedInsumos] = useState(dataSelected);
    const [total, setTotal] = useState(0);


    /**
     * @description: seleccionar o deseleccionar un insumo
     * @param {object} newInsumo - insumo
     */
    const toogleCheck = (newInsumo) => {

        // Actualizar insumos
        setInsumos(
            updateItemInArrayById(insumos, newInsumo)
        )

        // actualizar insumos seleccionados
        const updatedSelectedInsumos = newInsumo.checked
            ? [newInsumo, ...selectedInsumos]
            : removeItemFromArrayByProp('id', newInsumo.id, selectedInsumos);
        
        setSelectedInsumos( [...updatedSelectedInsumos] );
        setInLocalStorage('selected-insumos', updatedSelectedInsumos);


    }

    /**
     * @description: actualiza cantidad del insumo
     * @param {string} id 
     * @param {number} quantity 
     */
    const updateQuantityInSelectedInsumo = (id, quantity) => {

        setSelectedInsumos(
            updatingQuantity( id, quantity, selectedInsumos )
        );

    }

    /**
     * @description: selecciona todos los insumos
     * @param {array} insumos - arreglo de insumos
     */
    const selectingAllInsumos = (insumos) => {

        const allSelected = selectAllInsumos(insumos);

        setInsumos( allSelected );
        setSelectedInsumos([...allSelected]);
        setInLocalStorage('selected-insumos', allSelected);
    }
    
    /**
     * @description: deselecciona todos los insumos
     * @param {array} insumos - arreglo de insumos
     */
    const unSelectingAllInsumos = () => {
        
        const allUnSelected = unSelectAllInsumos( initialInsumos() );

        setInsumos( allUnSelected );
        setSelectedInsumos( [] );
        setInLocalStorage( 'selected-insumos', [] );
    };

    /**
     * @description: Busca insumos usando el titulo
     * @param {string} searchVal - criterio de busqueda
     */
    const searchingInsumos = (searchVal) => {

        if(searchVal === '' || searchVal === null || searchVal === undefined) {

            setInsumos( initialInsumos() );
        } else {

            const result = onSearch( insumos, searchVal );
            setInsumos( result );
        }

    }

    /**
     * @description: Busca insumos usando los labels
     * @param {string} searchVal - criterio de buscqueda
     */
    const filteringInsumos = (searchVal) => {

        if(searchVal === '' || searchVal === null || searchVal === undefined) {

            setInsumos( initialInsumos() );
        } else {

            const result = onFilter( insumos, searchVal );
            setInsumos( result );
        }

    }

    const addingNewInsumo = async (newInsumo) => {

        try {

            await setNewInsumo( newInsumo );
            setInsumos( i => [{ ...newInsumo }, ...i] );

        } catch(err) {
            console.log('Erro al crear insumo', err);
        }

    }

    const deletingInsumo = async (insumoId) => {

        await deleteInsumo(insumoId);

        setInsumos(
            removeItemFromArrayByProp('id', insumoId, insumos)
        );
    }

    // Total de insumos seleccionados => price * quantity
    useEffect( () => {

        setTotal(
            updateTotal( selectedInsumos )
        );
        
    }, [selectedInsumos])

    useEffect( () => {

        setInsumos(initInsumos);

    }, [initInsumos])

    const insumoContextProps = {
        addingNewInsumo,
        deletingInsumo,
        selectingAllInsumos,
        filteringInsumos,
        insumos,
        unSelectingAllInsumos,
        searchingInsumos,
        selectedInsumos,
        total,
        toogleCheck,
        updateQuantityInSelectedInsumo,
    }

    return (
        <InsumoContext.Provider
            value={ insumoContextProps }
        >
            { children }

        </InsumoContext.Provider>
    )
}