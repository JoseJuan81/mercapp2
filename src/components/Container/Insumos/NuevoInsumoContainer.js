import React from 'react';

import { PaginaNuevoInsumo } from '../../../Pages/Insumos/PaginaNuevoInsumo';
import { PaginaLoadingPaginaNuevoInsumo } from '../../../Pages/loading/PaginaLoadingPaginaNuevoInsumo';

import { NuevoInsumoMenuMobile } from '../../Menus/NuevoInsumoMenuMobile';

export const NuevoInsumoContainer = () => {

    return (
        <div className="layout__page">
            <PaginaNuevoInsumo />

            <NuevoInsumoMenuMobile />
        </div>
    )
}
