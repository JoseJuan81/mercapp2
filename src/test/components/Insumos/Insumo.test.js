import { mount } from 'enzyme';
import { Insumo } from '../../../components/Insumos/Insumo';
import { InsumoContext } from '../../../context/InsumoContext';

describe('Pruebas sobre archivo Insumo', () => {

    const INSUMO = {
        id: 1,
        title: 'Insumo de prueba',
        price: 34,
        currency: 'S/.',
        labels: ['etiqueta 1', 'etiqueta 2'],
        checked: false
    }

    const updateQuantityInSelectedInsumo = jest.fn();

    const wrapper = mount(
        <InsumoContext.Provider
            value={{
                updateQuantityInSelectedInsumo,
            }}
        >

            <Insumo
                { ...INSUMO }
            />

        </InsumoContext.Provider>
    )

    const INSUMO2 = {
        ...INSUMO,
        checked: true
    }

    const wrapper2 = mount(
        <InsumoContext.Provider
            value={{
                updateQuantityInSelectedInsumo,
            }}
        >

            <Insumo
                { ...INSUMO2 }
            />

        </InsumoContext.Provider>
    )

    test('Se debe mostrar correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('Se debe renderizar el title correctamente', () => {

        const dt = wrapper.find('dt');

        expect( dt.text().trim() ).toBe(INSUMO.title);

    })

    test('Se deben renderizar los labels correctamente', () => {

        const li = wrapper.find('li');

        expect( li.length ).toEqual(INSUMO.labels.length);
        expect( li.at(0).text().trim() ).toEqual(INSUMO.labels[0]);
        expect( li.at(1).text().trim() ).toEqual(INSUMO.labels[1]);
    })

    test('Se deben mostrar los componentes de cantidad, precio y total', () => {
        
        expect( wrapper2 ).toMatchSnapshot();

        const input = wrapper2.find('input');
        expect( input.prop('value') ).toBe(1);

        const dt = wrapper2.find('dt.text-lg.text-warmGray-400.font-semibold');
        const price = dt.childAt(1);
        expect( price.text().trim() ).toBe( INSUMO2.price.toString() );

        const dtTotal = wrapper2.find('dt.text-2xl.text-lime-500.font-bold');
        const total = dtTotal.childAt(2);
        expect( total.text().trim() ).toBe( INSUMO2.price.toString() );
    })

    test('Presionar boton para aumentar cantidad', () => {

        const addBtn = wrapper2.find( '[data-jest="addQuantity"]' );
        addBtn.simulate('click', { stopPropagation(){} });

        expect( updateQuantityInSelectedInsumo ).toHaveBeenCalled();

        const inputVal = wrapper2.find('input').prop('value');
        expect( inputVal ).toBe(2);
    })

    test('Presionar boton para disminuir cantidad', () => {

        const minusBtn = wrapper2.find( '[data-jest="minusQuantity"]' );
        minusBtn.simulate('click', { stopPropagation(){} });

        expect( updateQuantityInSelectedInsumo ).toHaveBeenCalled();

        const inputVal = wrapper2.find('input').prop('value');
        expect( inputVal ).toBe(1);
    })

    test('Presionar boton para disminuir cantidad 10 veces y el resultado debe ser 1', () => {

        const minusBtn = wrapper2.find( '[data-jest="minusQuantity"]' );
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });
        minusBtn.simulate('click', { stopPropagation(){} });

        expect( updateQuantityInSelectedInsumo ).toHaveBeenCalled();

        const inputVal = wrapper2.find('input').prop('value');
        expect( inputVal ).toBe(1);
    })

    test('Total de cantidad 10 * precio debe ser 340', () => {

        const addBtn = wrapper2.find( '[data-jest="addQuantity"]' );
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });
        addBtn.simulate('click', { stopPropagation(){} });

        expect( updateQuantityInSelectedInsumo ).toHaveBeenCalled();

        const inputVal = wrapper2.find('input').prop('value');
        expect( inputVal ).toBe(10);

        const totalDt = wrapper2.find('dt.text-2xl.text-lime-500.font-bold');
        const total = totalDt.childAt(2);
        expect( total.text().trim() ).toBe( String(INSUMO2.price * 10) );
    })
})