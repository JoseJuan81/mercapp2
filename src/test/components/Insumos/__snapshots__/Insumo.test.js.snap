// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre archivo Insumo Se debe mostrar correctamente 1`] = `
<div
  className="
                duration-200
                rounded-lg false
                border border-solid border-warmGray-300
            "
>
  <dl
    className="flex p-2 overflow-hidden relative"
  >
    <div
      className="flex flex-auto"
    >
      <dt
        className="
                duration-200
                flex-auto
                text-2xl font-bold text-warmGray-400
                pr-6
            "
      >
        Insumo de prueba
      </dt>
    </div>
    <div
      className="
                absolute right-0
                pr-2
                bg-warmGray-100
                flex
                transform translate-x-24
                border border-solid border-white
                rounded-l-full
                duration-300
            "
    >
      <button
        className="
                    bg-white
                    icon-circle-left
                    flex items-center justify-center
                    text-warmGray-500
                    text-2xl
                    transform rotate-0
                    w-10
                    false
                    border border-solid border-white
                    duration-300
                "
        onClick={[Function]}
        type="button"
      />
      <div
        className=" grid grid-cols-2 gap-2 "
      >
        <button
          className=" icon-pencil px-2 text-2xl text-blue-500 "
        />
        <button
          className=" icon-bin2 px-2 text-2xl text-rose-500 "
        />
      </div>
    </div>
  </dl>
  <div
    className="
                        rounded-br-lg rounded-bl-lg
                        bg-warmGray-100
                        p-1
                    "
  >
    <ul
      className="flex flex-wrap"
    >
      <li
        className="
                            bg-white
                            text-xs text-warmGray-800
                            m-1 p-1
                        "
        key="etiqueta 1-0-function random() { [native code] }"
      >
        etiqueta 1
      </li>
      <li
        className="
                            bg-white
                            text-xs text-warmGray-800
                            m-1 p-1
                        "
        key="etiqueta 2-1-function random() { [native code] }"
      >
        etiqueta 2
      </li>
    </ul>
  </div>
</div>
`;

exports[`Pruebas sobre archivo Insumo Se deben mostrar los componentes de cantidad, precio y total 1`] = `
<div
  className="
                duration-200
                rounded-lg shadow-xl
                border border-solid border-lime-400
            "
>
  <dl
    className="flex p-2 overflow-hidden relative"
  >
    <div
      className="flex flex-auto"
    >
      <dt
        className="
                duration-200
                flex-auto
                text-2xl font-bold text-lime-600
                pr-6
            "
      >
        Insumo de prueba
      </dt>
    </div>
    <div
      className="
                absolute right-0
                pr-2
                bg-warmGray-100
                flex
                transform translate-x-24
                border border-solid border-white
                rounded-l-full
                duration-300
            "
    >
      <button
        className="
                    bg-white
                    icon-circle-left
                    flex items-center justify-center
                    text-warmGray-500
                    text-2xl
                    transform rotate-0
                    w-10
                    false
                    border border-solid border-white
                    duration-300
                "
        onClick={[Function]}
        type="button"
      />
      <div
        className=" grid grid-cols-2 gap-2 "
      >
        <button
          className=" icon-pencil px-2 text-2xl text-blue-500 "
        />
        <button
          className=" icon-bin2 px-2 text-2xl text-rose-500 "
        />
      </div>
    </div>
  </dl>
  <div
    className="
                        rounded-br-lg rounded-bl-lg
                        bg-lime-100
                        p-1
                    "
  >
    <ul
      className="flex flex-wrap"
    >
      <li
        className="
                            bg-white
                            text-xs text-lime-700
                            m-1 p-1
                        "
        key="etiqueta 1-0-function random() { [native code] }"
      >
        etiqueta 1
      </li>
      <li
        className="
                            bg-white
                            text-xs text-lime-700
                            m-1 p-1
                        "
        key="etiqueta 2-1-function random() { [native code] }"
      >
        etiqueta 2
      </li>
    </ul>
  </div>
  <div
    className="flex justify-between items-center py-2 pr-4"
  >
    <div
      className="flex mx-2"
    >
      <button
        className=" icon-minus bg-gray-100 text-gray-500 w-12 h-12 "
        data-jest="minusQuantity"
        onClick={[Function]}
      />
      <input
        className=" mx-1 w-10 h-12 text-center text-xl font-medium text-warmGray-500 "
        onChange={[Function]}
        type="number"
        value={1}
      />
      <button
        className=" icon-plus bg-lime-50 text-lime-500 w-12 h-12 "
        data-jest="addQuantity"
        onClick={[Function]}
      />
    </div>
    <dt
      className=" text-lg text-warmGray-400 font-semibold "
    >
      <small
        className="text-xs font-light"
      >
        S/.
      </small>
      <span>
        34
      </span>
    </dt>
    <dt
      className=" text-2xl text-lime-500 font-bold "
    >
      <small
        className="text-xs font-normal mr-1"
      >
        total:
      </small>
      <small
        className="text-base font-light"
      >
        S/.
      </small>
      <span>
        34
      </span>
    </dt>
  </div>
</div>
`;
